builder/config_reader.go

```diff
@@ -45,18 +24,25 @@ type Config struct {
 	Lifecycle   LifecycleConfig     `toml:"lifecycle"`
 }
 
+// BuildpackCollection is a list of BuildpackConfigs
+type BuildpackCollection []BuildpackConfig
+
+// BuildpackConfig details the configuration of a Buildpack
 type BuildpackConfig struct {
 	dist.BuildpackInfo
 	dist.ImageOrURI
+	IsPackage bool
```

Comment:
There is a lot of risk and complexity introduced in making something mutable/stateful like this. Preferably this wouldn't live here and if necessary should be information that lives outside of this struct and is passed to whatever needs it.